overcast/tests/runner/test_main.py

```diff
@@ -682,6 +729,59 @@ def status(self):
         self.dr.record_resource.assert_any_call('port', 'nicuuid2')
         self.dr.record_resource.assert_any_call('server', 'serveruuid')
 
+
+    @mock.patch('overcast.runner.DeploymentRunner.create_port')
+    @mock.patch('overcast.runner.DeploymentRunner.get_nova_client')
+    @mock.patch('overcast.runner.DeploymentRunner.get_neutron_client')
+    @mock.patch('overcast.runner.DeploymentRunner.get_cinder_client')
+    @mock.patch('overcast.runner.time')
+    def test_create_node_with_map_boot_from_image(self, time, get_cinder_client, get_neutron_client, get_nova_client, create_port):
```

Comment:
There's far too much duplicated code in these three tests.

Try to put all the common code into a new method. Only methods that start with `test_` are run during the unit test run, so you can just call it e.g. `_test_create_node`. Make it take some extra arguments that makes it test the different scenarios. At a glance, it seems like you could simply pass in three variables: a dict that would be used to extend the node info, another dict representing the expected bdm dict being passed in and a string/NoneType denoting the expected image being passed to nc.servers.create.

Does that make sense?